prompt2model/dataset_retriever/dataset_index_file/retrieve_dataset_info_with_configs.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import annotations  # noqa FI58
+
+import argparse
+import gc
+import json
+import multiprocessing
+import sys
+import threading
+import time
+from collections.abc import MutableMapping
+from pathlib import Path
+from typing import Any
+
+import datasets
+import requests
+
+EXCLUDED_TAGS = [
+    "arxiv",
+    "region",
+    "license",
+    "size_categories",
+    "language_creators",
+]
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--index", type=int, default=100)
+
+parser.add_argument("--num_processes", type=int, default=4)
+
+
+def get_dataset_validity(dataset_name, max_retries=5):
+    """Get the list of loadable datasets from HuggingFace with backoff strategy."""
+    API_URL = f"https://datasets-server.huggingface.co/is-valid?dataset={dataset_name}"
```

Comment:
This is a local variable, so it should be lower case.
```suggestion
    api_url = f"https://datasets-server.huggingface.co/is-valid?dataset={dataset_name}"
```